docs/protocol.md

```diff
@@ -1,211 +1,371 @@
+
+## Overview
+
+This sequence diagram shows the flow of messages and values in the SecureDrop
+Protocol. The yellow boxes correspond to sections in the specification below
+that describe how these values are constructed and consumed.
+
+```mermaid
+sequenceDiagram
+
+participant Source
+
+box News Organization
+participant Server
+participant Journalist
+participant Newsroom
+end
+
+participant FPF
+
+Note over Newsroom, FPF: 2. Newsroom setup
+activate FPF
+activate Newsroom
+Newsroom ->> FPF: NRsig,pk := newsroom's signing key
+FPF ->> Newsroom: σFPF := FPF's signature
+deactivate FPF
+activate Server
+
+Note over Journalist, Server: 3.1. Journalist enrollment
+activate Journalist
+Journalist ->> Server: J{sig,fetch,dh},pk := journalist's long-term keys
+Newsroom ->> Server: σNR := newsroom's signature
+deactivate Newsroom
+
+Note over Journalist, Server: 3.2. Setup and periodic replenishment<br>of n ephemeral keys
+loop forall n:
+Journalist ->> Server: J{edh,ekem,epke},pk := journalist's ephemeral keys<br>σJ := journalist's signature
+end
+
+Note over Source: 4. Source setup
+Source ->> Source: passphrase
+
+alt Source → Journalist
+Note over Source, Server: 5. Source fetches keys and verifies<br>their authenticity
+activate Source
+Source ->> Server: request keys for newsroom
+Server ->> Source: NRsig,pk<br>σFPF
+loop forall journalists J:
+Server ->> Source: J{sig,fetch,dh},pk<br>σNR<br>J{edh,ekem,epke},pk<br>σJ
+end
+
+Note over Source, Server: 6. Source submits a message
+loop forall journalists J:
+Source ->> Server: C := message ciphertext<br>Z := public key<br>X := Diffie-Hellman share
+end
+
+Note over Server, Journalist: 7. Journalist fetches message IDs
+Journalist ->> Server: request messages
+loop forall n  messages:
+Server ->> Journalist: Q0...n := public keys<br>cid0...n := encrypted message IDs
+end
+
+Note over Server, Journalist: 8. Journalist fetches and decrypts a message
+Journalist ->> Server: id := decrypted message ID
+Server ->> Journalist: C
+
+else Journalist → Source
+Note over Server, Journalist: 9. Journalist replies to a source
+Journalist ->> Server: C' := message ciphertext<br>Z' := public key<br>X' := Diffie-Hellman share
+
+Note over Source, Server: 7. Source fetches message IDs
+Source ->> Server: request messages
+loop forall n  messages:
+Server ->> Source: Q'0...n := public keys<br>cid'0...n := encrypted message IDs
+end
+
+Note over Source, Server: 10. Source fetches and decrypts a message
+Source ->> Server: id' := decrypted message ID
+Server ->> Source: C'<br>X'
+end
+
+deactivate Source
+deactivate Journalist
+deactivate Server
+```
+
+## Keys
+
+In the table below:
+
+> For keys, we use the notation $X_{A,B}$, where $X$ represents the key owner
+> ($`X \in \{NR, J, S\}`$ [for newsroom, journalist, and source, respectively]),
+> $A$ represents the key's usage ($`A \in \{pke,sig,fetch,dh\}`$), and is prefixed
+> with an "e" if the key is ephemeral. $B$ indicates whether the component is
```

Comment:
For clarity, we could distinguish between ephemeral (the private key never gets stored) and one-time (the private key gets discarded after a user). Signal and some paper do that, I did not in my original readme and I think Luca followed that. As we improve language, it could be useful to make the distinction. So journalist receiving keys would be one-time, and per message keys would be ephemeral.

Reply:
Oh, good idea.  Filed in #77 as a self-contained revision.